src/v2/Components/v2/ArtworkFilter/ArtworkFilters/__tests__/PriceRangeFilter.jest.tsx

```diff
@@ -24,18 +25,93 @@ describe("PriceRangeFilter", () => {
     return <PriceRangeFilter />
   }
 
-  it("updates context on filter change", done => {
-    const wrapper = getWrapper() as any
-    act(() => {
-      wrapper
-        .find("Radio")
-        .filterWhere(n => n.text() === "$25k – $50k")
-        .simulate("click")
-
-      setTimeout(() => {
-        expect(context.filters.priceRange).toEqual("25000-50000")
-        done()
-      }, 0)
-    })
+  it("renders the options", () => {
+    const wrapper = getWrapper()
+    const options = wrapper.find("Radio")
+
+    expect(options).toHaveLength(6)
+  })
+
+  it("updates the filter on select", async () => {
+    const wrapper = getWrapper()
+    const option = wrapper
+      .find("Radio")
+      .filterWhere(n => n.text() === "$25K – $50K")
+
+    option.simulate("click")
+    await flushPromiseQueue()
+
+    expect(context.filters.priceRange).toEqual("25000-50000")
+  })
+
+  it("toggles the custom input", async () => {
+    const wrapper = getWrapper()
+
+    expect(wrapper.find(Input)).toHaveLength(0)
+
+    wrapper
+      .find("button")
+      .filterWhere(n => n.text() === "Show custom price")
+      .simulate("click")
+
+    expect(
+      wrapper.find("button").filterWhere(n => n.text() === "Hide custom price")
+    ).toHaveLength(1)
+
+    expect(wrapper.find(Input)).toHaveLength(2)
+  })
+
+  it("updates the input values when the radio selected option updates", async () => {
+    const wrapper = getWrapper()
+    const options = wrapper.find("Radio")
+
+    wrapper
+      .find("button")
+      .filterWhere(n => n.text() === "Show custom price")
+      .simulate("click")
+
+    expect(wrapper.find(Input).first().html()).toContain('value="*"')
+    expect(wrapper.find(Input).last().html()).toContain('value="*"')
+
+    options.first().simulate("click")
+    await flushPromiseQueue()
+
+    expect(wrapper.find(Input).first().html()).toContain('value="50000"')
+    expect(wrapper.find(Input).last().html()).toContain('value="*"')
+
+    options.last().simulate("click")
+    await flushPromiseQueue()
+
+    expect(wrapper.find(Input).first().html()).toContain('value="0"')
+    expect(wrapper.find(Input).last().html()).toContain('value="1000"')
+  })
+
+  // TODO: Unable to trigger this changes correctly
+  it.skip("updates the filter when the custom input is applied", async () => {
```

Comment:
I can't figure out how to get these inputs to actually trigger valid change events... Greatly dislike Enzyme.

Reply:
I don't think enzyme has a sponsor anymore opensource-wise; what would you use if you could choose?

Reply:
I found it a bit awkward to get used to but probably just [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/), since [it's what's "officially" recommended](https://reactjs.org/docs/test-utils.html#overview) anyway

![](https://static.damonzucconi.com/_capture/azcTfOs4.png)

Reply:
Lets discuss this in the next FE practice meeting -- in an app as large as force, I don't think it'd be so bad to switch this up a bit, if its worth it long-term.